record/internal.go

```diff
@@ -0,0 +1,371 @@
+// Copyright 2017 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package record
+
+import (
+	"errors"
+	"hash"
+	"hash/crc32"
+	"math"
+	"os"
+	"path/filepath"
+	"sync"
+
+	"github.com/prometheus/tsdb/chunkenc"
+	"github.com/prometheus/tsdb/chunks"
+	"github.com/prometheus/tsdb/fileutil"
+	"github.com/prometheus/tsdb/labels"
+)
+
+var (
+	// ErrOutOfOrderSample is returned if an appended sample has a
+	// timestamp smaller than the most recent sample.
+	ErrOutOfOrderSample = errors.New("out of order sample")
+
+	// ErrNotFound is returned if a looked up resource was not found.
+	ErrNotFound = errors.New("not found")
+
+	// ErrAmendSample is returned if an appended sample has the same timestamp
+	// as the most recent sample but a different value.
+	ErrAmendSample = errors.New("amending sample")
+)
+
+// The table gets initialized with sync.Once but may still cause a race
+// with any other use of the crc32 package anywhere. Thus we initialize it
+// before.
+var castagnoliTable *crc32.Table
```

Comment:
I don't think this is used.

Reply:
it's used by `NewCRC32`, which is used by tombstones in the record package